routes/profile.js

```diff
@@ -89,19 +89,21 @@ async function getProfileData(profileName, geoids, compare, db) {
   const queryBuilder = getQueryBuilder(profileName);
 
   // get data from postgres
-  const [rawProfileData, rawPreviousProfileData, rawCompareProfileData] = await Promise.all([
+  const [rawProfileData, rawCompareProfileData, rawPreviousProfileData, rawPreviousCompareProfileData] = await Promise.all([
     db.query(queryBuilder(profileName, geoids)),
-    db.query(queryBuilder(profileName, geoids, /* is previous */ true)),
     db.query(queryBuilder(profileName, [compare])),
+    db.query(queryBuilder(profileName, geoids, /* is previous */ true)),
+    db.query(queryBuilder(profileName, [compare], /* is previous */ true)),
   ]);
 
   // Instantiate DataProcessors to process query results
   const profileData = new DataProcessor(rawProfileData, profileName, isAggregate).process();
-  const previousProfileData = new DataProcessor(rawPreviousProfileData, profileName, isAggregate, /* isPrevious */ true).process();
   const compareProfileData = new DataProcessor(rawCompareProfileData, profileName, /* isAggregate */ false).process();
+  const previousProfileData = new DataProcessor(rawPreviousProfileData, profileName, isAggregate, /* isPrevious */ true).process();
+  const previousCompareProfileData = new DataProcessor(rawPreviousCompareProfileData, profileName, isAggregate, /* isPrevious */ true).process();
 
   // add previousProfileData and compareProfileData row objects into profileData row objects
-  join(profileData, previousProfileData, compareProfileData);
+  join(profileData, compareProfileData, previousProfileData, previousCompareProfileData);
 
```

Comment:
the other thing in the old code that didn't really make sense is that we are doing loops twice, 
join has a `for` loop, then `.map` is another loop, even though the `doChangeCalculations` and  `doDifferenceCalculations` can happen in the `for` loop within `join`